ddl/ddl_db_change_test.go

```diff
@@ -281,3 +282,61 @@ func (t *testExecInfo) execSQL(idx int) error {
 	}
 	return nil
 }
+
+// TestUpdateOrDelete tests whether the correct columns is used in PhysicalIndexScan's ToPB function.
+func (s *testStateChangeSuite) TestUpdateOrDelete(c *C) {
+	sqls := make([]string, 2)
+	sqls[0] = "delete from t where c2 = 'a'"
+	sqls[1] = "update t set c2 = 'c2_update' where c2 = 'a'"
```

Comment:
Will this update use index scan?

Reply:
Yes. 
The delete SQL can't use `index` in a statement directly, so these two statements use `drop state table` to have the index scan.

Reply:
We can add index hint to update statement.
```
"update t use index(c2) set c2 = 'c2_update' where c2 = 'a'"
```

Reply:
So how to deal with `delete` statement?